languagetool-language-modules/nl/src/main/java/org/languagetool/tagging/nl/DutchTagger.java

```diff
@@ -168,14 +188,27 @@ public List<AnalyzedTokenReadings> tag(List<String> sentenceTokens) {
           List<String> parts = compoundAcceptor.getParts(word);
           if (parts.size() == 2) {
             String part1 = parts.get(0);
-            List<AnalyzedTokenReadings> part2ReadingsList = tag(Collections.singletonList(parts.get(1)));
-            if (part2ReadingsList.size() > 0) {
+            String part2 = parts.get(1);
+            List<AnalyzedTokenReadings> part2ReadingsList = tag(Collections.singletonList(part2));
+            if (!part2ReadingsList.isEmpty()) {
               AnalyzedTokenReadings part2Readings = part2ReadingsList.get(0);
               String part1lc = part1.toLowerCase();
               for (AnalyzedToken part2Reading : part2Readings) {
-                if (part2Reading.getPOSTag() != null && part2Reading.getPOSTag().contains("ZNW")) {
-                  //System.out.println("Adding " + word + " with postag " + part2Reading.getPOSTag() + ", part2 has lemma " + part2Reading.getLemma());
-                  l.add(new AnalyzedToken(word, part2Reading.getPOSTag(), part1lc + part2Reading.getLemma()));
+                if (part2Reading.getPOSTag() != null && part2Reading.getPOSTag().startsWith("ZNW")) {
+                  // if word2 is contained in these lists, just give it the tag and exit the loop
+                  if (alwaysNeedsHet.contains(part2)){
+                    l.add(new AnalyzedToken(word, "ZNW:EKV:HET", part1lc + part2Reading.getLemma()));
+                    break;
+                  } else if (alwaysNeedsDe.contains(part2)){
+                    l.add(new AnalyzedToken(word, "ZNW:EKV:DE_", part1lc + part2Reading.getLemma()));
+                    break;
+                  } else if (alwaysNeedsMrv.contains(part2)){
+                    l.add(new AnalyzedToken(word, "ZNW:MRV:DE_", part1lc + part2Reading.getLemma()));
```

Comment:
To avoid duplication, please call sth. like `l.add(new AnalyzedToken(word, tag, part1lc + part2Reading.getLemma()));` at the end if the if/elseif and in the if/elseif only set the value of `tag`.